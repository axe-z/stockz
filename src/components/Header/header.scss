.header {
	position: relative;
	width: 100vw;
	height: 120px;
	background: rgb(233, 233, 233);
	span{
		margin-left: -5px;
		font-size: 11px;
	}
	h1 {
		font-family: helBold;
		padding: 40px 0 20px 20px;

	}
	.current {
		position: absolute;
		top:0;
		right: 0;

		width: 150px;
		transform:  translateX(150px);
		height: 120px;
		background: rgb(87, 74, 0);
		transition: transform 500ms ease;
		transition-delay: 300ms;
		div.headerInfo {
			margin-top: 15px;
			margin-left: 20px;
			p {
				color: rgba(255, 255, 255, 0.44);
				font-size: 14px;
				margin-top: 10px;
			}
			h6 {
				color: white;
				font-family: helMed;
				font-size: 15px;

			}
			.name {
				font-size: 20px;
				line-height: 0.5;
				letter-spacing: -1px;
				font-family: helBold;
			}
			.date {
				font-size: 11px;
			}
		}
	}
}