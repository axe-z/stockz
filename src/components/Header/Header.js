import React from 'react';
import './header.scss';

const transHead = {
	transform: 'translateX(0)'
}
const transBack = {
	transform: 'translateX(150px)'
}

const Header = ({stockName}) => {
	// console.log(stockName)
	return (
		<div>
			<div className="header">
				<h1>stockz <span>Â®</span></h1>

				<div className="current" style={stockName ? transHead : transBack }>
					<div className="headerInfo">
					 <h6 className="name">{stockName && stockName[0]}</h6>
					 <p>LogMeIn</p>
					 <h6 className="valeur">{stockName && stockName[1] + ' $'}</h6>
					 <p>last updated:</p>
					 <h6 className="date">{stockName && stockName[3]}</h6>
				  </div>
				</div>
			</div>
		</div>
	);
};

export default Header;