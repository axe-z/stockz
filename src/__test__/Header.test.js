import React from 'react';
import { mount, shallow, render } from 'enzyme';

import Header from '../components/Header/Header'



describe('Header va loader', () => {
  it('devrait partir en debug', () => {
    const component = shallow(<Header debug />);

    expect(component).toMatchSnapshot();
	});

	it('devrait renderer sans props', () => {
		const component = shallow(<Header/>);

		expect(component).toMatchSnapshot();
	});
});