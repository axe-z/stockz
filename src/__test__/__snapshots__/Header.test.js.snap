// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header va loader devrait partir en debug 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="header"
      >
        <h1>
          stockz 
          <span>
            ®
          </span>
        </h1>
        <div
          className="current"
          style={
            Object {
              "transform": "translateX(150px)",
            }
          }
        >
          <div
            className="headerInfo"
          >
            <h6
              className="name"
            />
            <h6
              className="valeur"
            />
            <p>
              last updated:
            </p>
            <h6
              className="date"
            />
          </div>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            stockz 
            <span>
              ®
            </span>
          </h1>,
          <div
            className="current"
            style={
              Object {
                "transform": "translateX(150px)",
              }
            }
          >
            <div
              className="headerInfo"
            >
              <h6
                className="name"
              />
              <h6
                className="valeur"
              />
              <p>
                last updated:
              </p>
              <h6
                className="date"
              />
            </div>
          </div>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "stockz ",
              <span>
                ®
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "stockz ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "®",
              },
              "ref": null,
              "rendered": "®",
              "type": "span",
            },
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="headerInfo"
            >
              <h6
                className="name"
              />
              <h6
                className="valeur"
              />
              <p>
                last updated:
              </p>
              <h6
                className="date"
              />
            </div>,
            "className": "current",
            "style": Object {
              "transform": "translateX(150px)",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h6
                  className="name"
                />,
                <h6
                  className="valeur"
                />,
                <p>
                  last updated:
                </p>,
                <h6
                  className="date"
                />,
              ],
              "className": "headerInfo",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "name",
                },
                "ref": null,
                "rendered": null,
                "type": "h6",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "valeur",
                },
                "ref": null,
                "rendered": null,
                "type": "h6",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "last updated:",
                },
                "ref": null,
                "rendered": "last updated:",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "date",
                },
                "ref": null,
                "rendered": null,
                "type": "h6",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="header"
        >
          <h1>
            stockz 
            <span>
              ®
            </span>
          </h1>
          <div
            className="current"
            style={
              Object {
                "transform": "translateX(150px)",
              }
            }
          >
            <div
              className="headerInfo"
            >
              <h6
                className="name"
              />
              <h6
                className="valeur"
              />
              <p>
                last updated:
              </p>
              <h6
                className="date"
              />
            </div>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              stockz 
              <span>
                ®
              </span>
            </h1>,
            <div
              className="current"
              style={
                Object {
                  "transform": "translateX(150px)",
                }
              }
            >
              <div
                className="headerInfo"
              >
                <h6
                  className="name"
                />
                <h6
                  className="valeur"
                />
                <p>
                  last updated:
                </p>
                <h6
                  className="date"
                />
              </div>
            </div>,
          ],
          "className": "header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "stockz ",
                <span>
                  ®
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "stockz ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "®",
                },
                "ref": null,
                "rendered": "®",
                "type": "span",
              },
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="headerInfo"
              >
                <h6
                  className="name"
                />
                <h6
                  className="valeur"
                />
                <p>
                  last updated:
                </p>
                <h6
                  className="date"
                />
              </div>,
              "className": "current",
              "style": Object {
                "transform": "translateX(150px)",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h6
                    className="name"
                  />,
                  <h6
                    className="valeur"
                  />,
                  <p>
                    last updated:
                  </p>,
                  <h6
                    className="date"
                  />,
                ],
                "className": "headerInfo",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "name",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h6",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "valeur",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h6",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "last updated:",
                  },
                  "ref": null,
                  "rendered": "last updated:",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "date",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h6",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Header va loader devrait renderer sans props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="header"
      >
        <h1>
          stockz 
          <span>
            ®
          </span>
        </h1>
        <div
          className="current"
          style={
            Object {
              "transform": "translateX(150px)",
            }
          }
        >
          <div
            className="headerInfo"
          >
            <h6
              className="name"
            />
            <h6
              className="valeur"
            />
            <p>
              last updated:
            </p>
            <h6
              className="date"
            />
          </div>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            stockz 
            <span>
              ®
            </span>
          </h1>,
          <div
            className="current"
            style={
              Object {
                "transform": "translateX(150px)",
              }
            }
          >
            <div
              className="headerInfo"
            >
              <h6
                className="name"
              />
              <h6
                className="valeur"
              />
              <p>
                last updated:
              </p>
              <h6
                className="date"
              />
            </div>
          </div>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "stockz ",
              <span>
                ®
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "stockz ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "®",
              },
              "ref": null,
              "rendered": "®",
              "type": "span",
            },
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="headerInfo"
            >
              <h6
                className="name"
              />
              <h6
                className="valeur"
              />
              <p>
                last updated:
              </p>
              <h6
                className="date"
              />
            </div>,
            "className": "current",
            "style": Object {
              "transform": "translateX(150px)",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h6
                  className="name"
                />,
                <h6
                  className="valeur"
                />,
                <p>
                  last updated:
                </p>,
                <h6
                  className="date"
                />,
              ],
              "className": "headerInfo",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "name",
                },
                "ref": null,
                "rendered": null,
                "type": "h6",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "valeur",
                },
                "ref": null,
                "rendered": null,
                "type": "h6",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "last updated:",
                },
                "ref": null,
                "rendered": "last updated:",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "date",
                },
                "ref": null,
                "rendered": null,
                "type": "h6",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="header"
        >
          <h1>
            stockz 
            <span>
              ®
            </span>
          </h1>
          <div
            className="current"
            style={
              Object {
                "transform": "translateX(150px)",
              }
            }
          >
            <div
              className="headerInfo"
            >
              <h6
                className="name"
              />
              <h6
                className="valeur"
              />
              <p>
                last updated:
              </p>
              <h6
                className="date"
              />
            </div>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              stockz 
              <span>
                ®
              </span>
            </h1>,
            <div
              className="current"
              style={
                Object {
                  "transform": "translateX(150px)",
                }
              }
            >
              <div
                className="headerInfo"
              >
                <h6
                  className="name"
                />
                <h6
                  className="valeur"
                />
                <p>
                  last updated:
                </p>
                <h6
                  className="date"
                />
              </div>
            </div>,
          ],
          "className": "header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "stockz ",
                <span>
                  ®
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "stockz ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "®",
                },
                "ref": null,
                "rendered": "®",
                "type": "span",
              },
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="headerInfo"
              >
                <h6
                  className="name"
                />
                <h6
                  className="valeur"
                />
                <p>
                  last updated:
                </p>
                <h6
                  className="date"
                />
              </div>,
              "className": "current",
              "style": Object {
                "transform": "translateX(150px)",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h6
                    className="name"
                  />,
                  <h6
                    className="valeur"
                  />,
                  <p>
                    last updated:
                  </p>,
                  <h6
                    className="date"
                  />,
                ],
                "className": "headerInfo",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "name",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h6",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "valeur",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h6",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "last updated:",
                  },
                  "ref": null,
                  "rendered": "last updated:",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "date",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h6",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
